src/api-client.js

```diff
@@ -0,0 +1,172 @@
+const core = require('@actions/core')
+const github = require('@actions/github')
+const hc = require('@actions/http-client')
+const { RequestError } = require('@octokit/request-error')
+const HttpStatusMessages = require('http-status-messages')
+
+// All variables we need from the runtime are loaded here
+const getContext = require('./context')
+
+// Mostly a lift from https://github.com/octokit/request.js/blob/bd72b7be53ab16a6c1c44be99eb73a328fb1e9e4/src/fetch-wrapper.ts#L151-L165
+// Minor revisions applied.
+function toErrorMessage(data) {
+  if (typeof data === 'string') return data
+
+  if (data != null && 'message' in data) {
+    if (Array.isArray(data.errors)) {
+      return `${data.message}: ${data.errors.map(JSON.stringify).join(', ')}`
+    }
+    return data.message
+  }
+
+  // Defer back to the caller
+  return null
+}
```

Comment:
Probably unnecessary since we're working with the Runtime API rather than the GitHub API here. 🤔